api/agent.go

```diff
@@ -162,8 +162,11 @@ func (a *Agent) MembersOpts(opts *QueryOptions) (*ServerMembers, error) {
 }
 
 // ForceLeave is used to eject an existing node from the cluster.
-func (a *Agent) ForceLeave(node string) error {
-	_, err := a.client.put("/v1/agent/force-leave?node="+node, nil, nil, nil)
+func (a *Agent) ForceLeave(node string, prune bool) error {
```

Comment:
Everything in this package is part of our public Go API, so we can't change the signature without breaking backwards compatibility. Unfortunately we'll need to add a new endpoint here. (I recommend something like `ForceLeaveWithOptions` and have an options struct as the second parameter, so that we can expand on that in the future if need be.)

Reply:
Many thanks for the backward compatibility advice, Tim! I have updated the code with new `ForceLeaveWithOptions` method.